docs/README.md

```diff
@@ -0,0 +1,126 @@
+# AWS Construct Library Design Guidelines
+
+The AWS Construct Library is a rich class library of CDK constructs which
```

Comment:
Feels like this content should be inside `api-design/README.md`?

At least, that's where you've pointed other hyperlinks that used to point to `DESIGN_GUIDELINES.md`, but they will now be missing this content.

Instead, I think this doc. should also be an index file with links to the other files.